packages/postman-to-openapi/src/convert.ts

```diff
@@ -0,0 +1,771 @@
+    replaceVars = false,
+    disabledParams = { includeQuery: false, includeHeader: false },
+    operationId = 'off',
+  } = options
+
+  let collectionFile: Record<string, any>
+  if (typeof collectionContent === 'string') {
+    collectionFile = JSON.parse(collectionContent)
+  } else {
+    collectionFile = collectionContent
+  }
+  if (replaceVars) {
+    collectionFile = replacePostmanVariables(collectionFile)
+  }
+  const postmanJson = collectionFile.collection || collectionFile
+  const { item: items = [], variable = [] } = postmanJson
+  const paths: Record<string, any> = {}
+  const domains = new Set<string>()
+  const tags: Record<string, any> = {}
+  const securitySchemes: Record<string, any> = {}
+
+  for (let i = 0; i < items.length; i++) {
+    let element = items[i]
+    while (element?.item != null) {
+      const { item = [], description: tagDesc = '' } = element
+      const tag = calculateFolderTag(element, folders)
+      const tagged = item.map((e: any) => ({ ...e, tag }))
+      tags[tag] = tagDesc
+      items.splice(i, 1, ...tagged)
+      element = tagged.length > 0 ? tagged.shift() : items[i]
+    }
+    if (element != null) {
+      const {
+        request = {},
+        name,
+        tag = defaultTag,
+        event: events,
+        response,
+      } = element
+      const {
+        url,
+        method,
+        body,
+        description: rawDesc,
+        header = [],
+        auth,
+      } = request
+      if (!url || !method) continue
+      const { path, query, protocol, host, port, valid, pathVars } =
+        scrapeURL(url)
+      if (valid) {
+        const summary = name.replace(/ \[([^[\]]*)\]/gi, '')
+        domains.add(calculateDomains(protocol, host, port))
+        const joinedPath = calculatePath(path, pathDepth)
+        if (!paths[joinedPath]) paths[joinedPath] = {}
+        const { description, paramsMeta } = descriptionParse(rawDesc)
+        paths[joinedPath][method.toLowerCase()] = {
+          tags: [tag],
+          summary,
+          ...calculateOperationId(operationId, name, summary),
+          ...(description ? { description } : {}),
+          ...parseBody(body, method),
+          ...parseOperationAuth(auth, securitySchemes, optsAuth),
+          ...parseParameters(
+            query,
+            header,
+            joinedPath,
+            paramsMeta,
+            pathVars,
+            disabledParams,
+          ),
+          ...parseResponse(response, events, responseHeaders),
+        }
+      }
+    }
+  }
+
+  const openApi: OpenAPIV3.Document = {
+    // TODO: We should output `3.1.0`, it’s probably enough to bump the version number here.
+    openapi: '3.0.0',
+    info: compileInfo(postmanJson, info),
+    ...parseExternalDocs(variable, externalDocs),
+    ...parseServers(domains, servers),
+    ...parseAuth(postmanJson, optsAuth, securitySchemes),
+    ...parseTags(tags),
+    paths,
+  }
+
+  return JSON.stringify(openApi, null, 4)
+}
+/* Helper Functions */
```

Comment:
Incorrect comment syntax for a function. If you want a module brake using the VSCode snippet `//`

Reply:
This module file is also getting quite long. Would suggest creating some separate module files to group some of these logically. All of the below functions need unit tests.